check_license/check-headers.sh

```diff
@@ -0,0 +1,186 @@
+		   -e 'TEST*' -e 'Makefile*' -e 'CMakeLists.txt$' -e '*\.cmake$' \
+		   -e '*\.link$' -e '*\.map$' -e '*\.Dockerfile$' -e 'LICENSE$' \
+		   -e '/common.inc$' -e '/match$' -e '/check_whitespace$' -e '/cppstyle$' | \
+	xargs)
+
+RV=0
+for file in $FILES ; do
+	if [ $VERBOSE -eq 1 ]; then
+		echo "Checking file: $file"
+	fi
+	# The src_path is a path which should be used in every command except git.
+	# git is called with -C flag so filepaths should be relative to SOURCE_ROOT
+	src_path="${SOURCE_ROOT}/$file"
+	[ ! -f $src_path ] && continue
+	# ensure that file is UTF-8 encoded
+	ENCODING=`file -b --mime-encoding $src_path`
+	iconv -f $ENCODING -t "UTF-8" $src_path > $TEMPFILE
+
+	if ! grep -q "SPDX-License-Identifier: $LICENSE" $src_path; then
+		echo >&2 "error: no $LICENSE SPDX tag in file: $src_path"
+		RV=1
+	fi
+
+	if [ $SHALLOW_CLONE -eq 0 ]; then
+		$GIT log --no-merges --format="%ai %aE" -- $file | sort > $TMP
+	else
+		# mark the grafted commits (commits with no parents)
+		$GIT log --no-merges --format="%ai %aE grafted-%p-commit" -- $file | sort > $TMP
+	fi
+
+	# skip checking dates for non-Intel commits
+	[[ ! $(tail -n1 $TMP) =~ "@intel.com" ]] && continue
+
+	# skip checking dates for new files
+	[ $(cat $TMP | wc -l) -le 1 ] && continue
+
+	# grep out the grafted commits (commits with no parents)
+	# and skip checking dates for non-Intel commits
+	grep -v -e "grafted--commit" $TMP | grep -e "@intel.com" > $TMP2
+
+	[ $(cat $TMP2 | wc -l) -eq 0 ] && continue
+
+	FIRST=`head -n1 $TMP2`
+	LAST=` tail -n1 $TMP2`
+
+	YEARS=$(sed '
+/.*Copyright (C) \+.*[0-9-]\+ Intel Corporation/!d
+s/.*Copyright (C) \([0-9]\+\)-\([0-9]\+\).*/\1-\2/
+s/.*Copyright (C) \([0-9]\+\).*/\1/' "$src_path")
+	if [ -z "$YEARS" ]; then
+		echo >&2 "No copyright years in $src_path"
+		RV=1
+		continue
+	fi
+
+	HEADER_FIRST=`echo $YEARS | cut -d"-" -f1`
+	HEADER_LAST=` echo $YEARS | cut -d"-" -f2`
+
+	COMMIT_FIRST=`echo $FIRST | cut -d"-" -f1`
+	COMMIT_LAST=` echo $LAST  | cut -d"-" -f1`
+	if [ "$COMMIT_FIRST" != "" -a "$COMMIT_LAST" != "" ]; then
+		if [[ -n "$COMMIT_FIRST" && -n "$COMMIT_LAST" ]]; then
+			if [[ $HEADER_FIRST -lt $COMMIT_FIRST ]]; then
+				HEADER_FIRST=$COMMIT_FIRST
+			fi
+			COMMIT_LAST=`date +%G`
+			if [[ $COMMIT_FIRST -eq $COMMIT_LAST ]]; then
+				NEW=$COMMIT_LAST
+			else
+				NEW=$COMMIT_FIRST-$COMMIT_LAST
+			fi
+			if [[ "$YEARS" == "$NEW" ]]; then
+				echo "No change needed: $YEARS"
+			else 
+				if [[ ${UPDATE_DATES} -eq 1 ]]; then
+					sed -i "s/Copyright ${YEARS}/Copyright ${NEW}/g" "${src_path}"
+				else
+					echo "$file:1: error: wrong copyright date: (is: $YEARS, should be: $NEW)" >&2
+					RV=1
+				fi
+			fi
+		fi
+	else
+		echo "error: unknown commit dates in file: $file" >&2
+		RV=1
+	fi
+done
+rm -f $TMP $TMP2 $TEMPFILE
+
+# check if error found
+if [ $RV -eq 0 ]; then
+	echo "Copyright headers are OK."
+else
+	echo "Error(s) in copyright headers found!" >&2
+fi
+exit $RV
```

Comment:
missing EOF (pls add "an extra enter" at the end)

Reply:
added